tools/rewriter/DetermineAbi.cpp

```diff
@@ -42,6 +42,13 @@ static std::vector<CAbiArgKind> classifyDirectType(const clang::Type &type,
           "unsupported scalar type (obj-C object, Clang block, or C++ member) found during ABI computation");
     }
   } else {
+    // Handle the case where we pass a struct directly in registers.
+    // The strategy here is to iterate through each field of the struct
+    // and record the ABI type each time we exit an eightbyte chunk.
+
+    // TODO I think this branch is dead code. We only call this in abiSlotsForArg
+    // with single-register values.
+
```

Comment:
This is definitely not dead code; it runs for the following types in our "structs" test: `s1`, `s2`, `s4`, `s7`. I believe this ends up being called whenever a struct type fits in a single register, but I don't know any particular guarantee Clang gives us around this, so this function does have some subsequent logic that may be dead.

Reply:
I see it now. I was getting "scalar" and "direct" confused I think.